llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -267,7 +271,11 @@ Value *VPTransformState::get(VPValue *Def, bool NeedsScalar) {
     return Data.VPV2Vector[Def];
 
   auto GetBroadcastInstrs = [this, Def](Value *V) {
-    bool SafeToHoist = Def->isDefinedOutsideLoopRegions();
+    bool SafeToHoist =
```

Comment:
It looks like there might be a bug in a function called `licm` in VPlanTransforms.cpp. Perhaps worth fixing that in this patch too?

Reply:
`licm` only operates on the loop region, so we should not visit recipes in the middle block (or successors of it) and using `isDefinedOutsideLoopRegions` should be sufficient?